transformer_engine/pytorch/tensor/_internal/float8_tensor_base.py

```diff
@@ -227,3 +227,10 @@ def update_usage(
         if not needs_data_transpose:
             self._transpose = None
             self._transpose_invalid = True
+
+    def get_usage(self) -> Dict[str, bool]:
+        """Get the usage of the tensor"""
+        return {
+            "rowwise": self._data is not None,
+            "columnwise": self._transpose is not None and not self._transpose_invalid,
+        }
```

Comment:
We should think more carefully about this. On Hopper, row-wise usage corresponds to non-transposed data and column-wise usage corresponds to transposed. On Blackwell, both row-wise and column-wise usage use the same non-transposed data. Logically, this should be something like:
```suggestion
    def get_usage(self) -> Dict[str, bool]:
        """Get the usage of the tensor"""
        usages = {"rowwise": self._data is not None}
        if is_non_tn_fp8_gemm_supported():
            usages["columnwise"] = self._data is not None
        else:
            usages["columnwise"] = self._transpose is not None and not self._transpose_invalid
        return usages
```